bin/network-monitoring/src/frontend.rs

```diff
@@ -0,0 +1,44 @@
+use axum::Router;
+use axum::response::Html;
+use axum::routing::get;
+use tower_http::services::ServeDir;
+
+use crate::status::{MonitoringConfig, SharedStatus};
+
+/// Runs the frontend server.
+///
+/// This function runs the frontend server that serves the dashboard and the status data.
+///
+/// # Arguments
+///
+/// * `shared_status` - The shared status of the network.
+/// * `config` - The configuration of the network.
+pub async fn run_frontend(shared_status: SharedStatus, config: MonitoringConfig) {
+    // build our application with routes
+    let app = Router::new()
+        // Serve static files from assets directory
+        .nest_service("/assets", ServeDir::new("bin/network-monitoring/assets"))
```

Comment:
Does this work when installed via `cargo install`? Because I don't think the `assets` folder will be bundled with unless `ServeDir` somehow embeds the assets into the binary.

I've typically forced this by embedding the files directly in the binary by using `include_str!` and friends, but then you can't use a directory approach.

See https://github.com/0xMiden/miden-faucet/issues/59 for some additional context.